ui/src/components/Drawer/components/Setting/index.vue

```diff
@@ -0,0 +1,318 @@
+      // 如果当前索引为最后一个，则跳转到第一个选项，否则向下移动
+      nextIndex = currentIndex === themeOptions.value.length - 1 ? 0 : currentIndex + 1;
+    }
+
+    currentTheme.value = themeOptions.value[nextIndex].value;
+    terminalSettingsStore.setDefaultTerminalConfig('theme', themeOptions.value[nextIndex].value);
+  }
+
+  if (event.key === 'Enter') {
+    handleUpdateTheme(currentTheme.value!);
+  }
+};
+
+/**
+ * @description 点击折叠按钮
+ * @param data
+ */
+const handleItemHeaderClick = (data: { name: string | number; expanded: boolean; event: MouseEvent }) => {
+  showLeftArrow.value = !data.expanded;
+};
+
+/**
+ * @description 移除在线用户
+ * @param userMeta
+ */
+const handlePositiveClick = (userMeta: OnlineUser) => {
+  const { socket, terminalId, sessionId } = currentTerminalConn.value;
+  socket?.send(
+    formatMessage(
+      terminalId,
+      FormatterMessageType.TERMINAL_SHARE_USER_REMOVE,
+      JSON.stringify({
+        session: sessionId,
+        user_meta: userMeta
+      })
+    )
+  );
+};
+
+/**
+ * @description 创建分享链接
+ * @param shareLinkRequest
+ */
+const handleCreateShareUrl = (shareLinkRequest: any) => {
+  const { socket, terminalId, sessionId } = currentTerminalConn.value;
+  socket?.send(
+    formatMessage(
+      terminalId,
+      FormatterMessageType.TERMINAL_SHARE,
+      JSON.stringify({
+        origin: origin.value,
+        session: sessionId,
+        users: shareLinkRequest.users,
+        expired_time: shareLinkRequest.expiredTime,
+        action_permission: shareLinkRequest.actionPerm
+      })
+    )
+  );
+};
+
+/**
+ * @description 搜索分享用户
+ * @param query
+ */
+const handleSearchShareUser = (query: string) => {
+  const { socket, terminalId } = currentTerminalConn.value;
+  socket?.send(formatMessage(terminalId, FormatterMessageType.TERMINAL_GET_SHARE_USER, JSON.stringify({ query })));
+};
+
+/**
+ * @description 写入命令
+ * @param command
+ */
+const handleWriteCommand = async (command: string) => {
+  const { terminal } = currentTerminalConn.value;
+
+  switch (command) {
+    case 'Stop':
+      terminal?.paste('\x03');
+      break;
+    case 'Save':
+      terminal?.paste('\x13');
+      break;
+    case 'Undo':
+      terminal?.paste('\x1A');
+      break;
+    case 'Paste':
+      terminal?.paste(await readText());
+      break;
+    case 'ArrowUp':
+      terminal?.paste('\x1b[A');
+      break;
+    case 'ArrowDown':
+      terminal?.paste('\x1b[B');
+      break;
+    case 'ArrowLeft':
+      terminal?.paste('\x1b[D');
+      break;
+    case 'ArrowRight':
+      terminal?.paste('\x1b[C');
+      break;
+  }
+};
+</script>
+
+<style scoped lang="scss">
+.n-form-item.n-form-item--top-labelled .n-form-item-label {
+  align-items: center;
+  padding: unset;
+}
+
+.n-collapse {
+  :deep(.n-collapse-item-arrow) {
+    display: none !important;
+  }
+
+  :deep(.n-collapse-item__content-inner) {
+    padding-top: 5px !important;
+  }
+}
+
+:deep(.n-form-item-label__text) {
+  width: 100%;
+}
+</style>
```

Comment:
There are no apparent issues with this code. It looks like it's being designed to create a UI for selecting themes for an XTerm-like terminal interface based on provided configuration options using Vue and TypeScript. The `storeToRefs` function is likely used to access state variables, which could be improved upon or replaced depending on specific needs.

It seems everything is functioning within the context of its intended implementation area but without extensive testing or verification against real-world conditions or edge cases may require additional testing outside of this environment. 

One potential optimization: Ensure the CSS classes and variable values are kept consistent throughout components to avoid potential changes breaking existing styles.